### ISSUE TITLE
Build action fails transiently in task airbyte-cdk:python:checkPython

### ISSUE THREAD

**Issue by tuliren** [airbytehq/airbyte#3590]
## Expected Behavior
Github build action should pass consistently.

## Current Behavior
The `format` task can fail transiently.

## Logs
https://github.com/airbytehq/airbyte/runs/2667103448

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':airbyte-cdk:python:checkPython'.
```

## Steps to Reproduce
1. Merge a PR and wait.

## Severity of the bug for you
Low

## Airbyte Version
`0.23.0-alpha`

## Connector Version (if applicable)
N/A

## Additional context
N/A

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200381284643274) by [Unito](https://www.unito.io)

**Comment by davinchia** [comment]
see https://github.com/xvik/gradle-use-python-plugin/issues/19 for more context

**Comment by davinchia** [comment]
I just ran into this. Leaving this logging here for future me. May be a red herring.

```
> Task :airbyte-integrations:connectors:source-file:checkPython FAILED
	 WARNING: Package(s) not found: virtualenv
Using python 3.7.10 from /opt/hostedtoolcache/Python/3.7.10/x64 (python3)
Using pip 21.1.2 from /opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pip (python 3.7)
[python] python3 -m pip install virtualenv==16.7.9 --user
	 Collecting virtualenv==16.7.9
	   Downloading virtualenv-16.7.9-py2.py3-none-any.whl (3.4 MB)
	 Installing collected packages: virtualenv
	 ERROR: Could not install packages due to an OSError: [Errno 17] File exists: '/home/runner/.local'
	 
	 WARNING: Package(s) not found: virtualenv

> Task :airbyte-integrations:bases:airbyte-protocol:checkPython
Using python 3.7.10 from /opt/hostedtoolcache/Python/3.7.10/x64 (python3)
Using pip 21.1.2 from /opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pip (python 3.7)
[python] python3 -m pip install virtualenv==16.7.9 --user
	 Collecting virtualenv==16.7.9
	   Downloading virtualenv-16.7.9-py2.py3-none-any.whl (3.4 MB)
	 Installing collected packages: virtualenv
	 Successfully installed virtualenv-16.7.9
Using virtualenv 16.7.9 (.venv)
[python] python3 -m virtualenv .venv
	 Using base prefix '/opt/hostedtoolcache/Python/3.7.10/x64'
	 New python executable in /home/runner/work/airbyte/airbyte/airbyte-integrations/bases/airbyte-protocol/.venv/bin/python3
	 Also creating executable in /home/runner/work/airbyte/airbyte/airbyte-integrations/bases/airbyte-protocol/.venv/bin/python
	 Installing setuptools, pip, wheel...
	 done.
Using python 3.7.10 from /home/runner/work/airbyte/airbyte/airbyte-integrations/bases/airbyte-protocol/.venv (.venv/bin/python)
Using pip 21.1.2 from /home/runner/work/airbyte/airbyte/airbyte-integrations/bases/airbyte-protocol/.venv/lib/python3.7/site-packages/pip (python 3.7)

2 actionable tasks: 2 executed
FAILURE: Build failed with an exception.


Publishing build scan...
* What went wrong:
Execution failed for task ':airbyte-integrations:connectors:source-file:checkPython'.
> Python call failed: python3 -m pip install virtualenv==16.7.9 --user
```

Workflow link: https://github.com/airbytehq/airbyte/runs/2701033944?check_suite_focus=true

**Comment by davinchia** [comment]
This should be fixed with #3362 . Will monitor and update.

**Comment by davinchia** [comment]
This looks good. Closing
